### ISSUE TITLE
Improvement: new feature iconRangeSame

### ISSUE THREAD

**Issue by viniciusjl** [wbotelhos/raty#135]
I've just added a new feature **iconRangeSame** that should be used in conjunction with **iconRange**.

The **iconRangeSame** is a _boolean_ feature (default is **false**), when it is set to **true**, all icons before the current one (icons at left of current icon) are shown as the current one.

This has the same effect as http://www.htmldrive.net/items/show/170/Colourful-rating-system-with-CSS3-And-jQuery (See also the issue #71 ).

Here is the diff:

```
diff -c jquery.raty.js.ORIG jquery.raty.js
*** jquery.raty.js.ORIG 2014-07-10 19:08:46.000000000 -0300
--- jquery.raty.js  2014-07-10 21:03:03.631117818 -0300
***************
*** 343,348 ****
--- 343,353 ----

      _fill: function(score) {
        var hash = 0;
+       if (this.opt.iconRangeSame && this.opt.iconRange) {
+         while (hash < this.opt.iconRange.length && this.opt.iconRange[hash].range < score) {
+           hash++;
+         }
+       }

        for (var i = 1; i <= this.stars.length; i++) {
          var
***************
*** 505,511 ****
      },

      _setIcon: function(star, icon) {
!       star[this.opt.starType === 'img' ? 'src' : 'className'] = this.opt.path + icon;
      },

      _setTarget: function(target, score) {
--- 510,517 ----
      },

      _setIcon: function(star, icon) {
!       if (star)
!         star[this.opt.starType === 'img' ? 'src' : 'className'] = this.opt.path + icon;
      },

      _setTarget: function(target, score) {
***************
*** 526,532 ****
            integer = parseInt(Math.ceil(score), 10),
            star    = this.stars[integer - 1];

!         star.title = methods._getHint.call(this, score, evt);
        }
      },

--- 532,539 ----
            integer = parseInt(Math.ceil(score), 10),
            star    = this.stars[integer - 1];

!         if (star)
!           star.title = methods._getHint.call(this, score, evt);
        }
      },

***************
*** 732,737 ****
--- 739,745 ----
      halfShow     : true,
      hints        : ['bad', 'poor', 'regular', 'good', 'gorgeous'],
      iconRange    : undefined,
+     iconRangeSame: false,
      mouseout     : undefined,
      mouseover    : undefined,
      noRatedMsg   : 'Not rated yet!',

Diff finished.  Thu Jul 10 21:18:46 2014
```

**Comment by OscarHMG** [comment]
Could you share the full .js file with the modifications? Thanks!

**Comment by wbotelhos** [comment]
Thanks @viniciusjl ! I was released in [v3.1.0](https://github.com/wbotelhos/raty/releases/tag/v3.1.0)
